python/cuml/decomposition/incremental_pca.py

```diff
@@ -0,0 +1,536 @@
+                                             verbose=verbose,
+                                             output_type=output_type)
+        self.batch_size = batch_size
+        self._param_names = ["n_components", "whiten", "copy", "batch_size"]
+
+    @with_cupy_rmm
+    def fit(self, X, y=None):
+        """Fit the model with X, using minibatches of size batch_size.
+        Parameters
+        ----------
+        X : array-like or sparse matrix, shape (n_samples, n_features)
+            Training data, where n_samples is the number of samples and
+            n_features is the number of features.
+        y : Ignored
+        Returns
+        -------
+        self : object
+            Returns the instance itself.
+        """
+
+        self._set_output_type(X)
+
+        self._sparse_model = True
+
+        self.n_samples_seen_ = 0
+        self._mean_ = .0
+        self.var_ = .0
+
+        if scipy.sparse.issparse(X) or cp.sparse.issparse(X):
+            X = _validate_sparse_input(X)
+        else:
+            X, n_samples, n_features, self.dtype = \
+                input_to_cuml_array(X, order='K',
+                                    check_dtype=[cp.float32, cp.float64])
+            X = X.to_output(output_type='cupy')
+
+        n_samples, n_features = X.shape
+
+        if self.batch_size is None:
+            self.batch_size_ = 5 * n_features
+        else:
+            self.batch_size_ = self.batch_size
+
+        for batch in _gen_batches(n_samples, self.batch_size_,
+                                  min_batch_size=self.n_components or 0):
+            X_batch = X[batch]
+            if cp.sparse.issparse(X_batch):
+                X_batch = X_batch.toarray()
+
+            self.partial_fit(X_batch, check_input=False)
+
+        return self
+
+    @with_cupy_rmm
+    def partial_fit(self, X, y=None, check_input=True):
+        """Incremental fit with X. All of X is processed as a single batch.
+        Parameters
+        ----------
+        X : array-like or sparse matrix, shape (n_samples, n_features)
+            Training data, where n_samples is the number of samples and
+            n_features is the number of features.
+        check_input : bool
+            Run check_array on X.
+        y : Ignored
+        Returns
+        -------
+        self : object
+            Returns the instance itself.
+        """
+        if check_input:
+            if scipy.sparse.issparse(X) or cp.sparse.issparse(X):
+                raise TypeError(
+                    "IncrementalPCA.partial_fit does not support "
+                    "sparse input. Either convert data to dense "
+                    "or use IncrementalPCA.fit to do so in batches.")
+
+            self._set_output_type(X)
+
+            X, n_samples, n_features, self.dtype = \
+                input_to_cuml_array(X, order='K',
+                                    check_dtype=[cp.float32, cp.float64])
+            X = X.to_output(output_type='cupy')
```

Comment:
For ease of use, I already record the output type a bit above https://github.com/rapidsai/cuml/pull/2615/files/f324329e7a3569d28e68798d78738ab07c2df8c8#diff-1126f22b757c9a25c74af9016b0812baR216. Since we earlier allowed PCA to accept sparse inputs, and this inherits from it, I modify the output back to what was expected. For example, here: https://github.com/rapidsai/cuml/pull/2615/files/f324329e7a3569d28e68798d78738ab07c2df8c8#diff-90c9ffafdb5e281b8e167baf3d487421R528

Reply:
Awesome! Do you mind putting a tiny comment in there?

Reply:
Of course, will do it!